system/jlib/jsocket.cpp

```diff
@@ -1164,6 +1164,7 @@ void CSocket::cancel_accept()
 #endif
     if (!in_accept) {
```

Comment:
I think I would shutdown() here and not differentiate between in_accept and not.